packages/gatsby/src/utils/develop-html-route.ts

```diff
@@ -0,0 +1,189 @@
+    } else {
+      splitLine = stackObject[0].split(`:`)
+    }
+    const splitLength = splitLine.length
+    filename = splitLine[splitLength - 3]
+    line = Number(splitLine[splitLength - 2])
+    row = Number(splitLine[splitLength - 1])
+  } catch (err) {
+    filename = ``
+    line = 0
+    row = 0
+  }
+  return {
+    filename: filename,
+    line: line,
+    row: row,
+  }
+}
+// Colors taken from Gatsby's design tokens
+// https://github.com/gatsbyjs/gatsby/blob/d8acab3a135fa8250a0eb3a47c67300dde6eae32/packages/gatsby-design-tokens/src/colors.js#L185-L205
+const colors = {
+  background: `fdfaf6`,
+  text: `452475`,
+  green: `137886`,
+  darkGreen: `006500`,
+  comment: `527713`,
+  keyword: `096fb3`,
+  yellow: `DB3A00`,
+}
+
+interface IParsedError {
+  filename: string
+  code: string
+  codeFrame: string
+  line: number
+  row: number
+  message: string
+  type: string
+  stack: [string]
+}
+
+// Code borrowed and modified from https://github.com/watilde/parse-error
+export const parseError = function (err, directory): IParsedError {
+  const stack = err.stack ? err.stack : ``
+  const stackObject = stack.split(`\n`)
+  const position = getPosition(stackObject)
+
+  // Remove the `/lib/` added by webpack
+  const filename = path.join(
+    directory,
+    // Don't need to use path.sep as webpack always uses a single forward slash
+    // as a path seperator.
+    ...position.filename.split(`/`).slice(2)
+  )
+
+  let code
+  try {
+    code = fs.readFileSync(filename, `utf-8`)
+  } catch (e) {
+    console.log(`original error`, err)
+    report.error(`Couldn't read the file ${filename}`, e)
+  }
+  const line = position.line
+  const row = position.row
+  ansiHTML.setColors({
+    reset: [colors.text, colors.background], // FOREGROUND-COLOR or [FOREGROUND-COLOR] or [, BACKGROUND-COLOR] or [FOREGROUND-COLOR, BACKGROUND-COLOR]
+    black: `aaa`, // String
+    red: colors.keyword,
+    green: colors.green,
+    yellow: colors.yellow,
+    blue: `eee`,
+    magenta: `fff`,
+    cyan: colors.darkGreen,
+    lightgrey: `888`,
+    darkgrey: colors.comment,
+  })
+  const codeFrame = ansiHTML(
+    codeFrameColumns(
+      code,
+      {
+        start: { line: line, column: row },
+      },
+      { forceColor: true }
+    )
+  )
+  const splitMessage = err.message ? err.message.split(`\n`) : [``]
+  const message = splitMessage[splitMessage.length - 1]
+  const type = err.type ? err.type : err.name
+  const data = {
+    filename: filename,
+    code,
+    codeFrame,
+    line: line,
+    row: row,
+    message: message,
+    type: type,
+    stack: stack,
+  }
+  return data
+}
+
+export const route = ({ app, program, store }): any =>
+  // Render an HTML page and serve it.
+  app.get(`*`, async (req, res, next) => {
+    if (isWebpackStatusPending()) {
+      res
+        .status(202)
+        .send(
+          `webpack isn't yet finished compiling code. Try refreshing once it's done.`
```

Comment:
Is this just a plain text string? Will this be handled on the client side or is this string displayed to the user?

Reply:
Yeah, just a raw string that'd be returned to the user. They'd need to keep refreshing.

On this however, @sidharthachatterjee and I assumed this would be needed but I couldn't ever hit this code path. It seems like the express server pauses while webpack is busy? Is this something you implemented? If so, we don't need this code.

Reply:
Yup, it _seems_ like us suspending webpack may be pausing express as well? (whether accidental or incidental) @ascorbic Do you recall if this is intended behaviour? 

In case, we haven't implemented this, we'll need a nicer looking page here (and a socket message to reload once ready)